src/workloads/sharding/ReshardCollectionWithIndexes.yml

```diff
@@ -0,0 +1,174 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+  This test expects the target cluster is created using ebs snapshot with 1 billion 1KB records 
+  and has 3 shards, named shard-00, shard-01, shard-02. The collection has 10 indexes including 
+  _id index, single-key index and compound index. The whole dataset is on 1 shard at the beginning.
+  ReshardCollection should use same-key resharding and use shardDistribution parameter to reshard 
+  into 2 shards then 3 shards. There will be random CRUD operations during resharding but should 
+  run at a very low rate.
+
+  The workload consists of 3 phases:
+    1. Turning off balancer and make the sharded collection exists on only 1 shard.
+    4. Running read and write operations on the collection while it is being resharded to 2 shards.
+    5. Running read and write operations on the collection while it is being resharded to 3 shards.
+    
+
+  The inserted documents have the following form:
+
+      {
+        _id: integer(default _id index),
+        shardKey: integer(for the shard key),
+        counter: integer(used for counting updates),
+        num1: integer(random generated between [1, 1000000000]),
+        num2: integer(random generated between [1, 1000000000]),
+        str1: string(length <= 20),
+        str2: string(length <= 20),
+        padding: string(random string of bytes ...)
+      }
+
+  The indexes are:
+      [
+        {_id: 1},
+        {shardKey: 1},
+        {counter: 1},
+        {num1: 1},
+        {num2: 1},
+        {str1: 1},
+        {str2: 1},
+        {shardKey: 1, counter:1},
+        {str1: 1, num1: 1},
+        {num2:1, str2: 1}
+      ]
+
+GlobalDefaults:
+  Nop: &Nop {Nop: true}
+
+  Database: &Database test
+  # Collection0 is the default collection populated by the MonotonicSingleLoader.
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+
+  DocumentCount: &DocumentCount 1_000_000_000  # for an approximate total of 1TB
+  NumChunks: &NumChunks 1                # start with only 1 shard
+  Timeout: &Timeout 72_000_000 # = 20 hours
+
+  ShardKeyValueMin: &ShardKeyValueMin 1
+  # We use 100x the number of chunks so that the $sample aggregation to choose the new split points
+  # has a wide range of distinct values to choose from. It could be even larger.
+  ShardKeyValueMax: &ShardKeyValueMax 1_000_000
+  ShardName0: &Shard0 shard-00
+  ShardName1: &Shard1 shard-01
+  ShardName2: &Shard2 shard-02
+
+  ReadRate: &ReadRate 100 per 1 second
```

Comment:
How were the read and write rate picked? Are they important? If not, why not?

Reply:
You have the reads and writes rate limited separately and running from different threads. That's probably the right thing for this test so you can see the impact on reads and writes separately. That said, there's an argument that could be made for having threads do a mixture of reads and writes (probably more representative), and that may behave differently coming from a shared thread (eg slow writes may impact following queries). 

I wanted to bring up the point and make sure you had considered it. Assuming you have considered it, you can close this comment thread.

Reply:
The read and write operations are not important here. We only want to have some CRUD operations here at a slow rate, so the rate is picked as some number I feel it's small.

Thanks for the idea to make this an argument, I didn't think about that because CRUD is not the main player in this test. I would leave it as it is now and make it tunable if we need it one day.

Reply:
That sounds good. Maybe add another comment here saying the values were picked to be small.